pkg/apis/wksprovider/controller/wksctl/machine_controller.go

```diff
@@ -300,15 +302,16 @@ func (a *MachineController) installNewBootstrapToken(ctx context.Context, ns str
 	if err != nil {
 		return nil, gerrors.Errorf("failed to create new bootstrap token %s/%s", ns, secret.ObjectMeta.Name)
 	}
-	s, err := a.clientSet.CoreV1().Secrets(ns).Create(ctx, secret, metav1.CreateOptions{})
+	err = a.client.Create(ctx, secret)
```

Comment:
Is it guaranteed that `secret` has the namespace populated correctly here?

Reply:
Yes, it is always populated as `kube-system`.